simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -563,6 +646,189 @@ simd<double, simd_abi::avx2_fixed_size<4>> condition(
+    return mask_type(_mm_cmpneq_ps(lhs.m_value, rhs.m_value));
+  }
+};
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> copysign(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a,
+    simd<float, simd_abi::avx2_fixed_size<4>> const& b) {
+  __m128 const sign_mask = _mm_set1_ps(-0.0);
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_xor_ps(_mm_andnot_ps(sign_mask, static_cast<__m128>(a)),
+                 _mm_and_ps(sign_mask, static_cast<__m128>(b))));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> abs(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a) {
+  __m128 const sign_mask = _mm_set1_ps(-0.0);
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_andnot_ps(sign_mask, static_cast<__m128>(a)));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> sqrt(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a) {
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_sqrt_ps(static_cast<__m128>(a)));
+}
+
+#ifdef __INTEL_COMPILER
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> cbrt(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a) {
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_cbrt_ps(static_cast<__m128>(a)));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> exp(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a) {
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_exp_ps(static_cast<__m128>(a)));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> log(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a) {
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_log_ps(static_cast<__m128>(a)));
+}
+
+#endif
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> fma(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a,
+    simd<float, simd_abi::avx2_fixed_size<4>> const& b,
+    simd<float, simd_abi::avx2_fixed_size<4>> const& c) {
+  return simd<float, simd_abi::avx2_fixed_size<4>>(_mm_fmadd_ps(
+      static_cast<__m128>(a), static_cast<__m128>(b), static_cast<__m128>(c)));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> max(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a,
+    simd<float, simd_abi::avx2_fixed_size<4>> const& b) {
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_max_ps(static_cast<__m128>(a), static_cast<__m128>(b)));
+}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<float, simd_abi::avx2_fixed_size<4>> min(
+    simd<float, simd_abi::avx2_fixed_size<4>> const& a,
+    simd<float, simd_abi::avx2_fixed_size<4>> const& b) {
+  return simd<float, simd_abi::avx2_fixed_size<4>>(
+      _mm_min_ps(static_cast<__m128>(a), static_cast<__m128>(b)));
+}
+
+[[nodiscard]] KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
```

Comment:
The `double` overload does not have that `[[nodiscard]]` attribute.

Reply:
`[[nodiscard]]` removed.